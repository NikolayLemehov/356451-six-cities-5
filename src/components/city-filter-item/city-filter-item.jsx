import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getCurrentCityName} from "../../store/selectors";
import {changeCityFilter} from "../../store/action";

const CityFilterItem = (props) => {
  const {city, currentCityName, onChangeCityFilter} = props;
  const isActiveCity = city === currentCityName;

  const handleFilterClick = (evt) => {
    evt.preventDefault();
    if (!isActiveCity) {
      onChangeCityFilter(city);
    }
  };

  return (
    <li className="locations__item">
      <a className={`locations__item-link tabs__item${
        isActiveCity ? ` tabs__item--active` : ``}`} href="#"
      onClick={handleFilterClick}
      >
        <span>{city}</span>
      </a>
    </li>
  );
};

CityFilterItem.propTypes = {
  city: PropTypes.string.isRequired,
  currentCityName: PropTypes.string.isRequired,
  onChangeCityFilter: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  currentCityName: getCurrentCityName(state),
});

const mapDispatchToProps = (dispatch) => ({
  onChangeCityFilter(cityName) {
    dispatch(changeCityFilter(cityName));
  },
});

export {CityFilterItem};
export default connect(mapStateToProps, mapDispatchToProps)(CityFilterItem);
