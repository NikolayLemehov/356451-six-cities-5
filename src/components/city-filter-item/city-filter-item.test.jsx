import React from "react";
import renderer from "react-test-renderer";
import {CityFilterItem} from "./city-filter-item";

const noop = () => {};
describe(`CityFilterItem should render correctly`, () => {
  it(`CityFilterItem should render correctly when city !== currentCityName`, () => {
    const tree = renderer
      .create(
          <CityFilterItem
            city={`City1`}
            currentCityName={`City2`}
            onChangeCityFilter={noop}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`CityFilterItem should render correctly when city === currentCityName`, () => {
    const tree = renderer
      .create(
          <CityFilterItem
            city={`City1`}
            currentCityName={`City1`}
            onChangeCityFilter={noop}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
