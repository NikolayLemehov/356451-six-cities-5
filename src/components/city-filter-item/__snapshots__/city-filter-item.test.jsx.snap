// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityFilterItem should render correctly CityFilterItem should render correctly when city !== currentCityName 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item"
    href="#"
    onClick={[Function]}
  >
    <span>
      City1
    </span>
  </a>
</li>
`;

exports[`CityFilterItem should render correctly CityFilterItem should render correctly when city === currentCityName 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
    onClick={[Function]}
  >
    <span>
      City1
    </span>
  </a>
</li>
`;
